sootup.callgraph/src/main/java/sootup/callgraph/CallGraph.java

```diff
@@ -180,8 +191,101 @@ boolean containsCall(
+  /**
+   * exports a call of the call graph to an edge in a dot file
+   *
+   * @param call the data of the call
+   * @return an edge defining the call in the dot file
+   */
+  default String toDotEdge(Call call) {
+    return "\""
+        + call.getSourceMethodSignature()
+        + "\"->\""
+        + call.getTargetMethodSignature()
+        + "\"[label=\""
+        + call.getLineNumber()
+        + "\"]";
+  }
+
+  /**
+   * This method converts the call graph object into dot format and write it to a string file.
+   *
+   * @return a String containing all edges of the call graph in the dot format
+   */
+  default String exportAsDot() {
+    String content =
+        getMethodSignatures().stream()
+            .flatMap(methodSignature -> callsFrom(methodSignature).stream())
+            .map(edge -> "\t" + toDotEdge(edge) + "\n")
+            .collect(Collectors.joining());
+    return "strict digraph ObjectGraph {\n" + content + "}";
+  }
+
+  /**
+   * This method converts the call graph object into dot format and writes it to a string file. The
+   * calls are sorted by the given Comparator for Call Objects.
+   *
+   * @param callComparator the comparator responsible for sorting the calls
+   * @return a String containing all edges of the call graph in the dot format
+   */
+  default String exportAsDot(Comparator<Call> callComparator) {
+    String content =
+        getMethodSignatures().stream()
+            .flatMap(methodSignature -> callsFrom(methodSignature).stream())
+            .sorted(callComparator)
+            .map(edge -> "\t" + toDotEdge(edge) + "\n")
+            .collect(Collectors.joining());
+    return "strict digraph ObjectGraph {\n" + content + "}";
+  }
+
+  /**
+   * This method converts the call graph object into dot format and writes it to a string file. The
+   * calls are sorted in order by the source and target method signature. These signatures are in
+   * order by the class name, the method name, and the parameter list
+   *
+   * @return a String containing all edges of the call graph in the dot format
+   */
+  default String exportAsDotSorted() {
+    // src class name
+    return exportAsDot(
+        Comparator.comparing(
+                (Call call) ->
+                    call.getSourceMethodSignature().getDeclClassType().getFullyQualifiedName())
+            // src method name
+            .thenComparing(call -> call.getSourceMethodSignature().getName())
+            // src parameter list
+            .thenComparing(call -> call.getSourceMethodSignature().getParameterTypes().toString())
+            // target class name
+            .thenComparing(
+                call -> call.getTargetMethodSignature().getDeclClassType().getClassName())
+            // target method name
+            .thenComparing(call -> call.getTargetMethodSignature().getName())
+            // target parameter list
+            .thenComparing(call -> call.getTargetMethodSignature().getParameterTypes().toString()));
+  }
```

Comment:
shall we provide just these Comparators + metntion them in the Javadoc to not clutter the interface?

Reply:
A related question: Should the export* etc. methods be part of the CallGraph interface at all? What about providing a separate DotExporter class instead?

(Unrelated side note: Hmm is building/returning a "big" in-memory String a good idea, in case of a "large" callgraph?)

Reply:
I can change it to a String List, but if we dont want to handle files I dont see a memory saving other way. 
Also this method can be build by the user themselves using all interface methods, in case of a more optimized version, the user can build it themselves